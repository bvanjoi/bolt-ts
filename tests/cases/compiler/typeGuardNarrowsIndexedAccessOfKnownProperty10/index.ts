// From `github.com/microsoft/TypeScript/blob/v5.8.2/tests/cases/compiler/typeGuardNarrowsIndexedAccessOfKnownProperty10.ts`, Apache-2.0 License

interface Foo { bar: any; }
const bar: { [id: string]: number } = {};

(foo: Foo) => {
	bar[id]++;
	//~^ ERROR: Block-scoped variable 'id' used before its declaration.
	const id = foo.bar;
}
