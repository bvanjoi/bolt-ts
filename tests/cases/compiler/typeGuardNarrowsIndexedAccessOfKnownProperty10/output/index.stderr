  × Block-scoped variable 'id' used before its declaration.
   ╭─[../../tests/cases/compiler/typeGuardNarrowsIndexedAccessOfKnownProperty10/index.ts:9:6]
 9 │ id
   · ──
   ╰────

Advice: 
  ☞ Block-scoped variable 'id' is defined here
    ╭─[../../tests/cases/compiler/typeGuardNarrowsIndexedAccessOfKnownProperty10/index.ts:11:8]
 11 │ id = foo.bar
    · ────────────
    ╰────
